test/plots/xyPlotTests.ts

```diff
@@ -2,268 +2,377 @@
+      });
+
+      it("can set the y property to a constant value", () => {
+        const constantY = 10;
+        assert.strictEqual(plot.y(constantY), plot, "setting the y property returns the plot");
+        assert.strictEqual(plot.y().accessor(null, 0, null), constantY, "returns an accessor returning the constant value");
+      });
 
-        plot.renderTo(svg);
+      it("can set the y property be based on data", () => {
+        const accessor = (d: any) => d * 10;
+        assert.strictEqual(plot.y(accessor), plot, "setting the y property returns the plot");
 
-        assert.deepEqual((<any>plot)._cachedDomainX, [5, 6], "The correct x domain is cached");
-        assert.deepEqual((<any>plot)._cachedDomainY, [6, 7], "The correct y domain is cached");
+        const testData = [4, 2, 3, 5, 6];
+        const dataset = new Plottable.Dataset(testData);
+        dataset.data().forEach((d, i) => {
+          assert.strictEqual(plot.y().accessor(d, i, dataset), accessor(d), "returns the input accessor");
+        });
       });
 
-      it("deferredRendering() caches domains properly setup before scales", () => {
-        xScale.domain([5, 6]);
-        yScale.domain([6, 7]);
-        plot.deferredRendering(true);
-        plot.x((d) => d.x, xScale);
-        plot.y((d) => d.y, yScale);
-        assert.deepEqual((<any>plot)._cachedDomainX, [null, null], "The x domain is still not cached until the timeout expires");
-        assert.deepEqual((<any>plot)._cachedDomainY, [null, null], "The y domain is still not cached until the timeout expires");
+      it("can set the y property be based on scaled data", () => {
+        const accessor = (d: any) => d * 10;
+        const scale = new Plottable.Scales.Linear();
+        assert.strictEqual(plot.y(accessor, scale), plot, "setting the y property returns the plot");
+
+        const testData = [4, 2, 3, 5, 6];
+        const dataset = new Plottable.Dataset(testData);
+        dataset.data().forEach((d, i) => {
+          const bindingAccessor = plot.y().accessor;
+          const bindingScale = plot.y().scale;
+          assert.strictEqual(bindingScale.scale(bindingAccessor(d, i, dataset)),
+            scale.scale(accessor(d)), "returns the input accessor and scale");
+        });
+      });
 
-        plot.renderTo(svg);
+      it("sets the range on the input scale if the plot has a height", () => {
+        const svg = TestMethods.generateSVG();
+        plot.anchor(svg);
+        plot.computeLayout();
 
-        assert.deepEqual((<any>plot)._cachedDomainX, [5, 6], "The x domain is still not cached until the timeout expires");
-        assert.deepEqual((<any>plot)._cachedDomainY, [6, 7], "The y domain is still not cached until the timeout expires");
+        const scale = new Plottable.Scales.Linear();
+        assert.strictEqual(plot.y(0, scale), plot, "setting the y property returns the plot");
+
+        assert.deepEqual(scale.range(), [plot.height(), 0], "range goes from height on scale");
```

Comment:
would this be different for category scale? if so can you include a test for that?